code/go/0chain.net/core/util/mpt_node_change.go

```diff
@@ -122,8 +125,21 @@ func (cc *ChangeCollector) UpdateChanges(ndb NodeDB, origin Sequence, includeDel
 	nodes := make([]Node, len(cc.Changes))
 	idx := 0
 	for _, c := range cc.Changes {
-		c.New.SetOrigin(origin)
-		keys[idx] = c.New.GetHashBytes()
+		if _, ok := c.New.(*LeafNode); ok && origin != c.New.GetOrigin() {
```

Comment:
Updating the origin would not break the references, so this logging code might not be needed.

Reply:
Origin of leaf nodes is included in hash bytes -- why wouldn't it break references?